src/client/Client.cc

```diff
@@ -9738,6 +9738,7 @@ void Client::unlock_fh_pos(Fh *f)
 int Client::uninline_data(Inode *in, Context *onfinish)
 {
   if (!in->inline_data.length()) {
+    // TODO - should we drop the client_lock here?
```

Comment:
@vshankar Can you weigh in here?

Reply:
Sorry - I wasn't following this PR closely. I'll take a look next week.

Reply:
although the `onfinish` context signals a condition variable on completion, its a no-op if there's no inline data to be uninlined, since the `onfinish` context is not handed over to an asynchronous network operation to signal is completion
so, IMO, we don't need to drop the `client_lock` here

Reply:
Thanks. Removed the comment.

Reply:
Also, onunline is now only done fro the write path AND is done synchronously, so all the async onunline bits that I had added in are gone.